src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -1176,20 +1176,22 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 
 	private async handleExplorerDrop(data: ElementsDragAndDropData<ExplorerItem, ExplorerItem[]>, target: ExplorerItem, originalEvent: DragEvent): Promise<void> {
 		const elementsData = FileDragAndDrop.getStatsFromDragAndDropData(data);
-		const distinctItems = new Set(elementsData);
+		const distinctItems = new Map(elementsData.map(element => [element, this.isCollapsed(element)]));
 
-		for (const item of distinctItems) {
-			if (this.isCollapsed(item)) {
+		for (const [item, collapsed] of distinctItems) {
+			if (collapsed) {
 				const nestedChildren = item.nestedChildren;
 				if (nestedChildren) {
 					for (const child of nestedChildren) {
-						distinctItems.add(child);
+						// if parent is collapsed, then the nested children is considered collapsed to operate as a group
+						// and skip collapsed state check since they're not in the tree
+						distinctItems.set(child, collapsed || this.isCollapsed(child));
```

Comment:
Since this is inside of the `if (collapsed)`, can `collapsed || this.isCollapsed(child)` be simplified to `true`?

Reply:
You are right :)
I have amended the commit